plugins/repository-hdfs/src/main/java/org/elasticsearch/repositories/hdfs/HdfsBlobContainer.java

```diff
@@ -75,8 +75,8 @@ public Boolean run(FileContext fileContext) throws IOException {
                     return fileContext.delete(new Path(path, blobName), true);
                 }
             });
-        } catch (FileNotFoundException ok) {
-            // behaves like Files.deleteIfExists
+        } catch (FileNotFoundException e) {
+            throw new IOException(e);
```

Comment:
It's already an `IOException`, it doesn't violate what should be the contract for this method.